### PR TITLE
Make workers count unbound 

### PR DESCRIPTION
**Pull request by mjudeikis**
Increase the default worker profile count on Create to 50

### CODE REVIEW

File: pkg/api/v20191231preview/openshiftcluster.go [hunk 1]
```diff
@@ -169,7 +169,7 @@ type WorkerProfile struct {
 	// The Azure resource ID of the worker subnet (immutable).
 	SubnetID string `json:"subnetId,omitempty"`
 
-	// The number of worker VMs.  Must be between 3 and 20 (immutable).
+	// The number of worker VMs.  Must be between 3 and 50 (immutable).
```
Comment (jim-minter) [inline pkg/api/v20191231preview/openshiftcluster.go:172]: I think we might be better off entirely removing `Must be ...` and `(immutable)` from our API definitions.

Comment (mjudeikis) [reply pkg/api/v20191231preview/openshiftcluster.go]: Including upper band value (as I did now)?

**Review by jim-minter — CHANGES_REQUESTED** [review]
I think my previous review wasn't clear.  I'd like to see:
* no comments related to validation (e.g. 3 <= valid <= 20; immutable) under pkg/api - so if we change our validations in the future, we don't have out of date comments in our clients
* for now at least, keep an upper bound on the number of worker VMs we can handle at install time.  If this moves from 20->50, I'm OK with that.

File: pkg/api/v20191231preview/openshiftcluster.go [hunk 2]
```diff
@@ -169,7 +169,7 @@ type WorkerProfile struct {
 	// The Azure resource ID of the worker subnet (immutable).
 	SubnetID string `json:"subnetId,omitempty"`
 
-	// The number of worker VMs.  Must be between 3 and 20 (immutable).
+	// The number of worker VMs.  Must greater than 3.
```
Comment (bennerv) [inline pkg/api/v20191231preview/openshiftcluster.go:172]: Should this instead be "Must be 3 or greater" or "Must be greater than 2"?

File: pkg/client/services/redhatopenshift/mgmt/2020-04-30/redhatopenshift/models.go [hunk 3]
```diff
@@ -780,6 +780,6 @@ type WorkerProfile struct {
 	DiskSizeGB *int32 `json:"diskSizeGB,omitempty"`
 	// SubnetID - The Azure resource ID of the worker subnet (immutable).
 	SubnetID *string `json:"subnetId,omitempty"`
-	// Count - The number of worker VMs.  Must be between 3 and 20 (immutable).
+	// Count - The number of worker VMs.  Must be greater than 3.
```
Comment (jim-minter) [inline pkg/client/services/redhatopenshift/mgmt/2020-04-30/redhatopenshift/models.go:783]: Looks like a `make generate` is needed

**Review by jim-minter — CHANGES_REQUESTED** [review]

File: pkg/api/v20191231preview/openshiftcluster.go (hunk 1 above)
Comment (jim-minter) [reply pkg/api/v20191231preview/openshiftcluster.go]: Yes, but I think we should do *all* the API definitions in one go in this PR.

File: pkg/client/services/redhatopenshift/mgmt/2020-04-30/redhatopenshift/models.go (hunk 3 above)
Comment (mjudeikis) [reply pkg/client/services/redhatopenshift/mgmt/2020-04-30/redhatopenshift/models.go]: yes... stale generate

**Review by jim-minter — APPROVED** [review]
@mjudeikis I've taken the liberty of pushing an additional commit.  If you're OK with it, please merge.  If you're not, please drop it :-)

**Pull request merged**